### PR TITLE
functionality 6

### PR DESCRIPTION
**Pull request by rkeynguyen**
Made an Arraylist to hold the result objects since there could be multiple products.

### CODE REVIEW

**Review by paveldanek — COMMENTED** [review]
Ricky, you've done a fine job! However, the way Brahma returns multiple results (and I copied it from A to Z, btw) is that he returns an iterator over a a list, not the list itself. It's safer and cheaper. Please, look at my getAllProducts() method of the GroceryStore. The ArrayList is private and after it's built, only an iterator over it is returned.

File: src/store/facade/GroceryStore.java [hunk 1]
```diff
@@ -639,4 +639,36 @@ public Result processOrder(Request request) {
...
 	}
+
+	/**
+	 * Changes the price of a product
+	 * 
+	 * @param s is the name string
+	 * @return a result code ArrayList that represents the outcome
+	 */
+	public ArrayList<Result> getProductInfo(String s) {
+		ArrayList<Result> result = new ArrayList<Result>();
```
Comment (carterclark) [inline src/store/facade/GroceryStore.java]: This method name seems to differ from the description given

File: src/ui/UserInterface.java [hunk 2]
```diff
@@ -488,7 +489,25 @@ public void changePrice() {
 
 	}
 
+	/**
+	 * Lists all products that start with a given name and displays their name, id,
+	 * price, stock in hand, and reorder level.
+	 */
 	public void getProductInfo() {
```
Comment (carterclark) [inline src/ui/UserInterface.java:496]: I think this is supposed to be for functionality 7

**Review by carterclark — COMMENTED** [review]
Logic looks good, could you change the description?

File: src/store/facade/GroceryStore.java [hunk 3]
```diff
@@ -639,4 +639,36 @@ public Result processOrder(Request request) {
...
+			}
+		}
+
+		// if its empty that means no products start with that name
+		if (result.isEmpty()) {
+			result.add(new Result());
+			result.get(0).setResultCode(Result.ACTION_FAILED);
+		}
+
+		return result;
```
Comment (carterclark) [inline src/store/facade/GroceryStore.java]: // ignore this Ricky, I'm testing github functions

```suggestion
		return result.iterator();
```

File: src/ui/UserInterface.java [hunk 4]
```diff
@@ -488,7 +489,25 @@ public void changePrice() {
...
+		String name = getString("Enter product's name: ");
+		ArrayList<Result> result = groceryStore.getProductInfo(name);
+
+		if (result.get(0).getResultCode() != Result.ACTION_SUCCESSFUL) {
+			System.out.println("Error: product does not exist");
```
Comment (carterclark) [inline src/ui/UserInterface.java:501]:
```suggestion
			System.out.println("Error: product does not exist"); yes baby
```

**Review by carterclark — COMMENTED** [review]
Hi

**Comment by rkeynguyen** [comment]
made the changes can you guys look over it again

**Review by carterclark — APPROVED** [review]

**Pull request merged**